src/full/Agda/TypeChecking/Rules/LHS.hs

```diff
@@ -1745,11 +1737,8 @@ disambiguateProjection h ambD@(AmbQ ds) b = do
           tryDisambiguate True fs r vs comatching $ \case
             ([]   , []      ) -> __IMPOSSIBLE__
             (err:_, []      ) -> throwError err
-            (_    , disambs@((d,a):_)) -> typeError . GenericDocError =<< vcat
-              [ "Ambiguous projection " <> prettyTCM d <> "."
-              , "It could refer to any of"
-              , nest 2 $ vcat $ map (prettyDisambProj . fst) disambs
-              ]
+            (_    , disambs@((d,a):_)) -> typeError . AmbiguousProjection d =<< (mapM (prettyDisambProj . fst) disambs)
```

Comment:
Can `AmbiguousProjection` get the names of the projections directly, and printing them is deferred to printing the error?